src/XRP/xrp-client.ts

```diff
@@ -20,14 +24,25 @@ class XRPClient implements XRPClientInterface {
    * The XRPClient will use gRPC to communicate with the given endpoint.
    *
    * @param grpcURL The URL of the gRPC instance to connect to.
+   * @param network The network this XRPClient is connecting to. Defaults to undefined. If set to undefined, Testnet is assumed.
    * @param useNewProtocolBuffers If `true`, then the new protocol buffer implementation from rippled will be used. Defaults to true.
    * @param forceWeb If `true`, then we will use the gRPC-Web client even when on Node. Defaults to false. This is mainly for testing and in the future will be removed when we have browser testing.
    */
   public constructor(
     grpcURL: string,
+    network: XRPLNetwork | undefined = undefined,
```

Comment:
I think this is a breaking change to the constructor since this changes the 2nd parameter.

Three options:
(1) Given this is a breaking change to the constructor anyway, if we want the `network` parameter to be required, then we could make it required right now.
(2) Move the `network` parameter to be last (after `forceWeb`). That way, existing users will continue to work without changes. (Non-breaking change)
(3) Use an [options object](https://rclayton.silvrback.com/easy-class-api-options-with-typescript-and-joi). This can be a breaking or non-breaking change, depending on whether you replace the existing parameters, or just add a new options object.

Reply:
Agree you're right. I'm going to just rip the band aid off. We're going to do a major semver anyway because we're moving to support rippled directly so we may as well just do this. 

Options objects are interesting - I think that this should be a required parameter but I'll consider this in the future if things keep getting more complicated.